specification/compute/resource-manager/Microsoft.Compute/preview/2019-05-01-preview/ScriptJob.json

```diff
@@ -0,0 +1,625 @@
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/scriptJobs/{jobName}": {
+      "get": {
+        "tags": [
+          "RunScript Job"
+        ],
+        "description": "Retrieves the job identified by the job name.",
+        "externalDocs": {
+          "url": "TBD"
+        },
+        "operationId": "ScriptJob_Get",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/VmNameParameter"
+          },
+          {
+            "$ref": "#/parameters/JobNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "$expand",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "The expand expression to apply on the operation."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Job"
+            }
+          },
+          "default": {
+            "description":"Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get RunScript Job": {
+            "$ref": "./examples/GetRunScriptJob.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "RunScript Job"
+        ],
+        "description": "Creates or updates a RunScript Job.",
+        "operationId": "ScriptJob_CreateOrUpdate",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/VmNameParameter"
+          },
+          {
+            "$ref": "#/parameters/JobNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/JobCreateParameters"
```

Comment:
Since this is `CreateOrUpdate` it's ideal if the body parameter can be the same time as what's returned from your `Get` operation as it makes it easier for clients to get/modify/put.  Is that possible here?

Reply:
The Job object in the Put request body is not the same as the one in Get request body.  The content that is used to create the job (which is in the body of the Put request) will not be returned in the Get.  This is because the typical Get of a Job is for its status - whether it Succeeded or Failed.  So, we don't want to unnecessarily return the content for every Get call.  We intend to implement a separate API to get the content.  That is why these are different.